bool enable
float64 angle
